### PR TITLE
Added package def for sinon

### PR DESCRIPTION
**Pull request by AGawrys**
Agawrys/P2X-321-flow-types-for-sinon-pkg

# ↪️ Pull Request

This PR adds `sinon` to flow-types

See also PR made to [sinon flow type](https://github.com/flow-typed/flow-typed/pull/3934) for relevant changes

## 🚨 Test instructions

Can test imports by adding` // flow strict enable` to files that use `sinon`

Also running ` yarn flow check` and seeing there are 0 errors

### CODE REVIEW

**Comment by AGawrys** [comment]
@wbinnssmith Sinon definition updated 😀

File: packages/core/integration-tests/test/hmr.js [hunk 1]
```diff
@@ -9,6 +9,7 @@ import {
   overlayFS,
   sleep,
 } from '@parcel/test-utils';
+// flowlint untyped-import:off
```
Comment (wbinnssmith) [inline packages/core/integration-tests/test/hmr.js:12]:
```suggestion
// flowlint-next-line untyped-import:off
```

File: packages/core/core/test/ParcelConfig.test.js [hunk 2]
```diff
@@ -1,4 +1,6 @@
-// @flow
+// @flow strict-local
+
+// flowlint untyped-import:off
```
Comment (wbinnssmith) [inline packages/core/core/test/ParcelConfig.test.js:3]: why is this needed?

File: packages/core/integration-tests/test/hmr.js [hunk 3]
```diff
@@ -23,7 +24,7 @@ async function closeSocket(ws: typeof WebSocket) {
   ws.close();
   await new Promise(resolve => (ws.onclose = resolve));
 }
-
+// flowlint-next-line unclear-type:off
```
Comment (wbinnssmith) [inline packages/core/integration-tests/test/hmr.js:27]: This should be resolvable once we have typings for the ws package :)

File: packages/core/integration-tests/test/hmr.js (hunk 1 above)
Comment (AGawrys) [reply packages/core/integration-tests/test/hmr.js:12]: The reason I didn't include the `next-line `here is that I wanted the rule applied for the next 4 lines, which all are untyped imports. the flowlint would only apply to the rest of the file, and not before so it wouldn't affect the previous imports.

File: packages/core/core/test/ParcelConfig.test.js (hunk 2 above)
Comment (AGawrys) [reply packages/core/core/test/ParcelConfig.test.js:3]: ahh mistake

File: packages/core/integration-tests/test/hmr.js (hunk 3 above)
Comment (AGawrys) [reply packages/core/integration-tests/test/hmr.js:27]: Do you suggest removing this?

File: packages/core/core/test/ParcelConfig.test.js (hunk 2 above)
Comment (wbinnssmith) [reply packages/core/core/test/ParcelConfig.test.js:3]: what remains untyped in this file?

File: packages/core/integration-tests/test/hmr.js (hunk 3 above)
Comment (wbinnssmith) [reply packages/core/integration-tests/test/hmr.js:27]: not now — when we add typings for ws :)

File: packages/core/integration-tests/test/hmr.js (hunk 1 above)
Comment (wbinnssmith) [reply packages/core/integration-tests/test/hmr.js:12]: Ah, got it. Personally I would suppress each import individually, as this suppresses the violation for any imports added after this line in the future. But this is fine to me, especially if we can get typings for the remaining packages and remove it soon.

thoughts @padmaia @mischnic?

Comment (mischnic) [reply packages/core/integration-tests/test/hmr.js:12]: > Personally I would suppress each import individually, as this suppresses the violation for any imports added after this line in the future.

👍

File: packages/core/package-manager/test/NodePackageManager.test.js [hunk 4]
```diff
@@ -1,5 +1,6 @@
-// @flow
+// @flow strict-local
 
+// flowlint untyped-import:off
```
Comment (wbinnssmith) [inline packages/core/package-manager/test/NodePackageManager.test.js:3]: same here

File: packages/core/integration-tests/test/hmr.js (hunk 1 above)
Comment (wbinnssmith) [reply packages/core/integration-tests/test/hmr.js:12]: Another option might be to group all the untyped imports and enable the lint rule afterwards with `// flowlint untyped-import:error`.

**Review by wbinnssmith — APPROVED** [review]
